service/s3/endpoints.go

```diff
@@ -206,3 +211,6590 @@ func finalizeClientEndpointResolverOptions(options *Options) {
+	//
+	// AWS::UseFIPS
+	UseFIPS *bool
+
+	// When true, use the dual-stack endpoint. If the configured endpoint does not
+	// support dual-stack, dispatching the request MAY return an error.
+	//
+	// Defaults to
+	// false if no value is provided.
+	//
+	// AWS::UseDualStack
+	UseDualStack *bool
+
+	// Override the endpoint used to send this request
+	//
+	// Parameter is
+	// required.
+	//
+	// SDK::Endpoint
+	Endpoint *string
+
+	// When true, force a path-style endpoint to be used where the bucket name is part
+	// of the path.
+	//
+	// Parameter is required.
+	//
+	// AWS::S3::ForcePathStyle
+	ForcePathStyle *bool
+
+	// When true, use S3 Accelerate. NOTE: Not all regions support S3
+	// accelerate.
+	//
+	// Defaults to false if no value is provided.
+	//
+	// AWS::S3::Accelerate
+	Accelerate *bool
+
+	// Whether the global endpoint should be used, rather then the regional endpoint
+	// for us-east-1.
+	//
+	// Defaults to false if no value is
+	// provided.
+	//
+	// AWS::S3::UseGlobalEndpoint
+	UseGlobalEndpoint *bool
+
+	// Internal parameter to use object lambda endpoint for an operation (eg:
+	// WriteGetObjectResponse)
+	//
+	// Parameter is required.
+	UseObjectLambdaEndpoint *bool
+
+	// Internal parameter to disable Access Point Buckets
+	//
+	// Parameter is required.
+	DisableAccessPoints *bool
+
+	// Whether multi-region access points (MRAP) should be disabled.
+	//
+	// Defaults to false
+	// if no value is provided.
+	//
+	// AWS::S3::DisableMultiRegionAccessPoints
+	DisableMultiRegionAccessPoints *bool
+
+	// When an Access Point ARN is provided and this flag is enabled, the SDK MUST use
+	// the ARN's region when constructing the endpoint instead of the client's
+	// configured region.
+	//
+	// Parameter is required.
+	//
+	// AWS::S3::UseArnRegion
+	UseArnRegion *bool
+}
+
+// ValidateRequired validates required parameters are set.
+func (p EndpointParameters) ValidateRequired() error {
+	if p.Accelerate == nil {
+		return fmt.Errorf("parameter Accelerate is required")
+	}
+
+	if p.DisableMultiRegionAccessPoints == nil {
+		return fmt.Errorf("parameter DisableMultiRegionAccessPoints is required")
+	}
+
+	if p.UseDualStack == nil {
+		return fmt.Errorf("parameter UseDualStack is required")
+	}
+
+	if p.UseFIPS == nil {
+		return fmt.Errorf("parameter UseFIPS is required")
+	}
+
+	if p.UseGlobalEndpoint == nil {
+		return fmt.Errorf("parameter UseGlobalEndpoint is required")
+	}
+
+	return nil
+}
+
+// EndpointResolverV2 provides the interface for resolving service endpoints.
+type EndpointResolverV2 interface {
+	// ResolveEndpoint attempts to resolve the endpoint with the provided options,
+	// returning the endpoint if found. Otherwise an error is returned.
+	ResolveEndpoint(ctx context.Context, options EndpointParameters) (
+		smithytransport.Endpoint, error,
+	)
+}
+
+// resolverV2 provides the implementation for resolving endpoints.
+type resolverV2 struct{}
+
+func NewDefaultEndpointResolverV2() *resolverV2 {
```

Comment:
**fix**: Type should be `EndpointResolverV2` interface not the concrete type right?